mirrortest.pp

```diff
@@ -0,0 +1,44 @@
+  package { "lvm2":
+    ensure => "installed"
+  }
+
+
+  #-----------------------------------------------------------------
+  # LVM (Logical Volume Manager) mirrors in puppet by Martin Wangel.
+  # Tested locally, and with Amazon EBS volumes on EC2 instances.
+  #-----------------------------------------------------------------
+  $lvm_file_system="ext4"              # ext4 or xfs, xfs can not be resized by this version of this puppet module
+  $vgname="mmvg"                       # name of Volume Group
+  $lvname="mmlv"                       # name of Logical Volume
+  $devices=["/dev/sdb1","/dev/sdb2"]   # Physical Devices
+  $lv_fs_size="2G"                     # Size of Logical Volume
+
+  physical_volume { [$devices]:
+      ensure => present
+  }
+  volume_group { "${vgname}":
+      ensure => present,
+      physical_volumes => $devices,
+      require =>  [ Physical_volume[$devices] ]
+  } ->
+  logical_volume { "${lvname}":
+      ensure => present,
+      volume_group => $vgname,
+      size => $lv_fs_size,
+      mirror => 1,
+      mirrorlog => core,
+      region_size => 4,
+      alloc => normal,
+      no_sync => 1
+  } ->
+  filesystem { "/dev/${vgname}/${lvname}":
+      ensure => present,
+      fs_type => $lvm_file_system,
+  } ->
+  mount { "lvm-${vgname}-${lvname}":
+    name    => "/u",
+    ensure  => mounted,
+    device  => "/dev/${vgname}/${lvname}",
+    fstype  => $lvm_file_system,
+    options => "defaults"
+  }
```

Comment:
I have replaced a new pull request for the old one because of merge
conflicts and rebase issues. The example is in the pull request message
because I don't know where to put an examples file exactly. There is only
one commit now (deba31aac1147ff48dc5e87812ed4c0dd532e893).

  Martin Wangel


On Fri, Feb 6, 2015 at 4:45 PM, Morgan Haskel <notifications@github.com>
wrote:

> In mirrortest.pp
> <https://github.com/puppetlabs/puppetlabs-lvm/pull/76#discussion_r24248405>
> :
>
> > +      mirrorlog => core,
> > +      region_size => 4,
> > +      alloc => normal,
> > +      no_sync => 1
> > +  } ->
> > +  filesystem { "/dev/${vgname}/${lvname}":
> > +      ensure => present,
> > +      fs_type => $lvm_file_system,
> > +  } ->
> > +  mount { "lvm-${vgname}-${lvname}":
> > +    name    => "/u",
> > +    ensure  => mounted,
> > +    device  => "/dev/${vgname}/${lvname}",
> > +    fstype  => $lvm_file_system,
> > +    options => "defaults"
> > +  }
>
> @mwangel <https://github.com/mwangel> you can create an examples
> directory in the module and add this file there if you'd like. Thanks!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/puppetlabs/puppetlabs-lvm/pull/76/files#r24248405>.
>



-- 
-----------------------
 Martin Wangel
 Iteego